src/lib.rs

```diff
@@ -158,11 +158,7 @@ pub fn try_cast_ref<A: Pod, B: Pod>(a: &A) -> Result<&B, PodCastError> {
   if align_of::<B>() > align_of::<A>() && (a as *const A as usize) % align_of::<B>() != 0 {
     Err(PodCastError::TargetAlignmentGreaterAndInputNotAligned)
   } else if size_of::<B>() == size_of::<A>() {
-    Ok(unsafe {
-      (a as *const A as *const B)
-        .as_ref()
-        .unwrap_or_else(|| core::hint::unreachable_unchecked())
-    })
+    Ok(unsafe { &*(a as *const A as *const B) })
```

Comment:
This also happened with .as_mut() elsewhere in this file FWIW but glad I'm not missing anything